app/store/modules/product/index.js

```diff
@@ -1,25 +1,78 @@
 // store/modules/product/index.js
 
 import productData from './mock/product';
+import api from './../../utils/api';
 
 export default {
   namespaced: true,
   state: {
     /**
      * @type {Object}   Stores product data
      */
-    product: productData
+    product: productData,
+
+    /**
+     * @type {Object}   Stores product information data
+     */
+    productInformation: {},
+
+    /**
+     * @type {Object} API error
+     */
+    apiError: null,
   },
   getters: {
     /**
      * Gets a product
      *
      * @param   {Object}  state   Current state
      *
-     * @returns  {Array}  product   Product data
+     * @returns  {Object}  product   Product data
+     */
+    getProduct: state => state.product,
+
+    /**
+     * Get the product information
+     *
+     * @param   {Object}  state   Current state
+     *
+     * @returns  {Object}  product   Product information data
+     */
+    getProductInformation: state => state.productInformation,
+  },
+  mutations: {
+    /**
+     * Sets the productinformation for the current state
+     * @param {Object} state Current state
+     * @param {Object} productInformation Product information data
      */
-    getProduct: state => state.product
+    setProductInformation(state, productInformation) {
+      state.productInformation = productInformation;
+    },
+
+    /**
+     * Handles an API failure
+     * @param {Object} state Current state
+     * @param {Object} error API Error that ocured
+     */
+    apiFailure(state, error) {
+      state.apiError = error;
+    }
+  },
+  actions: {
+    getProductInformation({ commit, state }) {
+      const payload = [{
```

Comment:
- [ ] This is only mock. Mark it like that with a TODO. Even better replace with a property and define the mock where this method is called.